extras/ds1307/ds1307.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Driver for DS1307 RTC
+ *
+ * Part of esp-open-rtos
+ * Copyright (C) 2016 Ruslan V. Uss <unclerus@gmail.com>
+ * BSD Licensed as described in the file LICENSE
+ */
+#ifndef EXTRAS_DS1307_H_
+#define EXTRAS_DS1307_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <time.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Squarewave frequency
+ */
+typedef enum _ds1307_squarewave_freq_t
+{
+    DS1307_1HZ = 0, //!< 1 Hz
+    DS1307_4096HZ,  //!< 4096 Hz
+    DS1307_8192HZ,  //!< 8192 Hz
+    DS1307_32768HZ  //!< 32768 Hz
+} ds1307_squarewave_freq_t;
+
+/**
+ * \brief Init RTC
+ * \param scl_pin GPIO pin for I2C SCL
+ * \param sda_pin GPIO pin for I2C SDA
+ */
+void ds1307_init(uint8_t scl_pin, uint8_t sda_pin);
```

Comment:
I think it was @ourairquality who pointed out that initialising I2C driver inside another driver might be inconvenient when you have several drivers in the system that use the same I2C.